File: rito-admin/src/main/java/com/stylefeng/guns/modular/guar/model/GuarAcctBsInfoFb.java
package com.stylefeng.guns.modular.guar.model;

import com.baomidou.mybatisplus.annotations.TableField;
import com.baomidou.mybatisplus.annotations.TableId;
import com.baomidou.mybatisplus.annotations.TableName;
import com.baomidou.mybatisplus.enums.IdType;
import lombok.Data;

import java.io.Serializable;
import java.util.Date;

/**
 * <p>
 * 企业担保-账户信息流水表
 * </p>
 *
 * @author wj123
 * @since 2019-06-19
 */
@TableName("GUAR_ACCT_BS_INFO_FB")
@Data
public class GuarAcctBsInfoFb implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * ID
     */
    @TableId(value = "ID", type = IdType.ID_WORKER)
    private Long id;
    /**
     * 账户类型
     */
    @TableField("ACCT_TYPE")
    private String acctType;
    /**
     * 账户标识码
     */
    @TableField("ACCT_CODE")
    private String acctCode;
    /**
     * 信息报告日期
     */
    @TableField("RPT_DATE")
    private Date rptDate;
    /**
     * 报告时点说明代码
     */
    @TableField("RPT_DATE_CODE")
    private String rptDateCode;
    /**
     * 债务人名称
     */
    @TableField("NAME")
    private String name;
    /**
     * 债务人身份标识类型
     */
    @TableField("ID_TYPE")
    private String idType;
    /**
     * 债务人身份标识号码
     */
    @TableField("ID_NUM")
    private String idNum;
    /**
     * 业务管理机构代码
     */
    @TableField("MNGMT_ORG_CODE")
    private String mngmtOrgCode;
    /**
     * 担保业务大类
     */
    @TableField("BUSI_LINES")
    private String busiLines;
    /**
     * 担保业务种类细分
     */
    @TableField("BUSI_DTIL_LINES")
    private String busiDtilLines;
    /**
     * 开户日期
     */
    @TableField("OPEN_DATE")
    private Date openDate;
    /**
     * 担保金额
     */
    @TableField("GUAR_AMT")
    private Long guarAmt;
    /**
     * 币种
     */
    @TableField("CY")
    private String cy;
    /**
     * 到期日期
     */
    @TableField("DUE_DATE")
    private Date dueDate;
    /**
     * 反担保方式
     */
    @TableField("GUAR_MODE")
    private String guarMode;
    /**
     * 其他还款保证方式
     */
    @TableField("OTH_REP_GUAR_WAY")
    private String othRepGuarWay;
    /**
     * 保证金百分比
     */
    @TableField("SEC_DEP")
    private Integer secDep;
    /**
     * 担保合同文本编号
     */
    @TableField("CTRCT_TXT_CODE")
    private String ctrctTxtCode;
    /**
     * 责任人个数
     */
    @TableField("RLT_REPYMT_NM")
    private Integer rltRepymtNm;
    /**
     * 抵质押合同个数
     */
    @TableField("CC_NM")
    private Integer ccNm;
    /**
     * 授信协议标识码
     */
    @TableField("MCC")
    private String mcc;
    /**
     * 账户状态
     */
    @TableField("ACCT_STATUS")
    private String acctStatus;
    /**
     * 在保余额
     */
    @TableField("LOAN_AMT")
    private Long loanAmt;
    /**
     * 余额变化日期
     */
    @TableField("REPAY_PRD")
    private Date repayPrd;
    /**
     * 五级分类
     */
    @TableField("FIVE_CATE")
    private String fiveCate;
    /**
     * 五级分类认定日期
     */
    @TableField("FIVE_CATE_ADJ_DATE")
    private Date fiveCateAdjDate;
    /**
     * 风险敞口
     */
    @TableField("RIEX")
    private Long riex;
    /**
     * 代偿（垫款）标志
     */
    @TableField("COMP_ADV_FLAG")
    private String compAdvFlag;
    /**
     * 账户关闭日期
     */
    @TableField("CLOSE_DATE")
    private Date closeDate;
    /**
     * 信息记录状态 
     */
    @TableField("STATUS")
    private String status;
    /**
     * 创建人用户主键ID
     */
    @TableField("CREATE_USER")
    private Long createUser;
    /**
     * 创建时间
     */
    @TableField("CREATE_TIME")
    private Date createTime;
    /**
     * 最后修改人员的用户ID
     */
    @TableField("UPDATE_USER")
    private Long updateUser;
    /**
     * 最后修改时间
     */
    @TableField("UPDATE_TIME")
    private Date updateTime;
    /**
     * 删除标志
     */
    @TableField("DELETE_FLAG")
    private String deleteFlag;


}

File: rito-admin/src/main/webapp/assets/modular/guar/guarAcctBsInfoFb/guarAcctBsInfoFb.js
/**
 * 账户信息流水表管理初始化
 */

layui.use(['layer', 'form', 'table', 'admin', 'ax'], function () {
    var $ = layui.$;
    var layer = layui.layer;
    var form = layui.form;
    var table = layui.table;
    var $ax = layui.ax;
    var admin = layui.admin;

    /**
     * 系统管理--消息管理
     */
    var guarAcctBsInfoFb = {
        tableId: "guarAcctBsInfoFbTable"    //表格id
    };

    /**
     * 初始化表格的列
     */
    guarAcctBsInfoFb.initColumn = function () {
        return [[
            {type: 'checkbox'},
            {type: 'numbers', title: '序号', width: 50},
            {title: 'ID', field: 'id', visible: true, align: 'center', valign: 'middle'},
            {title: '账户类型', field: 'acctType', visible: true, align: 'center', valign: 'middle'},
            {title: '账户标识码', field: 'acctCode', visible: true, align: 'center', valign: 'middle'},
            {title: '信息报告日期', field: 'rptDate', visible: true, align: 'center', valign: 'middle'},
            {title: '报告时点说明代码', field: 'rptDateCode', visible: true, align: 'center', valign: 'middle'},
            {title: '债务人名称', field: 'name', visible: true, align: 'center', valign: 'middle'},
            {title: '债务人身份标识类型', field: 'idType', visible: true, align: 'center', valign: 'middle'},
            {title: '债务人身份标识号码', field: 'idNum', visible: true, align: 'center', valign: 'middle'},
            {title: '业务管理机构代码', field: 'mngmtOrgCode', visible: true, align: 'center', valign: 'middle'},
            {title: '担保业务大类', field: 'busiLines', visible: true, align: 'center', valign: 'middle'},
            {title: '担保业务种类细分', field: 'busiDtilLines', visible: true, align: 'center', valign: 'middle'},
            {title: '开户日期', field: 'openDate', visible: true, align: 'center', valign: 'middle'},
            {title: '担保金额', field: 'guarAmt', visible: true, align: 'center', valign: 'middle'},
            {title: '币种', field: 'cy', visible: true, align: 'center', valign: 'middle'},
            {title: '到期日期', field: 'dueDate', visible: true, align: 'center', valign: 'middle'},
            {title: '反担保方式', field: 'guarMode', visible: true, align: 'center', valign: 'middle'},
            {title: '其他还款保证方式', field: 'othRepGuarWay', visible: true, align: 'center', valign: 'middle'},
            {title: '保证金百分比', field: 'secDep', visible: true, align: 'center', valign: 'middle'},
            {title: '担保合同文本编号', field: 'ctrctTxtCode', visible: true, align: 'center', valign: 'middle'},
            {title: '责任人个数', field: 'rltRepymtNm', visible: true, align: 'center', valign: 'middle'},
            {title: '抵质押合同个数', field: 'ccNm', visible: true, align: 'center', valign: 'middle'},
            {title: '授信协议标识码', field: 'mcc', visible: true, align: 'center', valign: 'middle'},
            {title: '账户状态', field: 'acctStatus', visible: true, align: 'center', valign: 'middle'},
            {title: '在保余额', field: 'loanAmt', visible: true, align: 'center', valign: 'middle'},
            {title: '余额变化日期', field: 'repayPrd', visible: true, align: 'center', valign: 'middle'},
            {title: '五级分类', field: 'fiveCate', visible: true, align: 'center', valign: 'middle'},
            {title: '五级分类认定日期', field: 'fiveCateAdjDate', visible: true, align: 'center', valign: 'middle'},
            {title: '风险敞口', field: 'riex', visible: true, align: 'center', valign: 'middle'},
            {title: '代偿（垫款）标志', field: 'compAdvFlag', visible: true, align: 'center', valign: 'middle'},
            {title: '账户关闭日期', field: 'closeDate', visible: true, align: 'center', valign: 'middle'},
            {title: '信息记录状态 ', field: 'status', visible: true, align: 'center', valign: 'middle'},
            {title: '创建人用户主键ID', field: 'createUser', visible: true, align: 'center', valign: 'middle'},
            {title: '创建时间', field: 'createTime', visible: true, align: 'center', valign: 'middle'},
            {title: '最后修改人员的用户ID', field: 'updateUser', visible: true, align: 'center', valign: 'middle'},
            {title: '最后修改时间', field: 'updateTime', visible: true, align: 'center', valign: 'middle'},
            {title: '删除标志', field: 'deleteFlag', visible: true, align: 'center', valign: 'middle'},
            {align: 'center', toolbar: '#tableBar', title: '操作', minWidth: 200}
        ]];
    };

    /**
     * 点击查询按钮
     */
    guarAcctBsInfoFb.search = function () {
        var queryData = {};
        queryData['condition'] = $("#condition").val();
        table.reload(guarAcctBsInfoFb.tableId, {where: queryData});
    };

    /**
     * 弹出添加账户信息流水表
     */
    guarAcctBsInfoFb.openAddguarAcctBsInfoFb = function () {
        window.location.href = Feng.ctxPath + '/guarAcctBsInfoFb/guarAcctBsInfoFb_add';

    };


    /**
     * 点击编辑账户信息流水表
     *
     * @param data 点击按钮时候的行数据
     */
    guarAcctBsInfoFb.onEditguarAcctBsInfoFb = function (data) {

        window.location.href = Feng.ctxPath + '/guarAcctBsInfoFb/guarAcctBsInfoFb_update?guarAcctBsInfoFbId=' + data.id;

    };


    /**
     * 点击删除账户信息流水表
     *
     * @param data 点击按钮时候的行数据
     */
    guarAcctBsInfoFb.onDeleteguarAcctBsInfoFb = function (data) {
        var operation = function () {
            var ajax = new $ax(Feng.ctxPath + "/guarAcctBsInfoFb/delete", function (data) {
                Feng.success("删除成功!");
                table.reload(guarAcctBsInfoFb.tableId);
            }, function (data) {
                Feng.error("删除失败!" + data.responseJSON.message + "!");
            });
            ajax.set("guarAcctBsInfoFbId", data.guarAcctBsInfoFbId);
            ajax.start();
        };
        Feng.confirm("是否删除账户信息流水表 " + data.title + "?", operation);
    };

    // 渲染表格
    var tableResult = table.render({
        elem: '#' + guarAcctBsInfoFb.tableId,
        url: Feng.ctxPath + '/guarAcctBsInfoFb/list',
        page: true,
        height: "full-98",
        cellMinWidth: 100,
        cols: guarAcctBsInfoFb.initColumn()
    });

    // 搜索按钮点击事件
    $('#btnSearch').click(function () {
        guarAcctBsInfoFb.search();
    });

    // 添加按钮点击事件
    $('#btnAdd').click(function () {
        guarAcctBsInfoFb.openAddguarAcctBsInfoFb();
    });

    // 工具条点击事件
    table.on('tool(' + guarAcctBsInfoFb.tableId + ')', function (obj) {
        var data = obj.data;
        var layEvent = obj.event;

        if (layEvent === 'edit') {
            guarAcctBsInfoFb.onEditguarAcctBsInfoFb(data);
        } else if (layEvent === 'delete') {
            guarAcctBsInfoFb.onDeleteguarAcctBsInfoFb(data);
        }
    });
});
